### PR TITLE
Fixing RunAsMe on Windows

### PR DESCRIPTION
**Pull request by fviale**
- KeyPairUtil : initialize SecureRandom statically for performance and test at initialization that it works
- TaskContextSerializer : set shared permissions on context file
- ExecuteForkedTaskInsideNewJvm : wait some time if the task context file cannot be delete and improve error messages
- ForkedJvmTaskExecutionCommandCreator : build the classpath using scheduler home and use wildcards to avoid command too long errors
- ForkedTaskExecutor : removed obsolete catch block and add taskid name to the process reader threads
- InProcessTaskExecutor : use task id to generate the node file instead of randomisation
- ProcessStreamReader : add name to the stream reader threads

### CODE REVIEW

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 6 issues

1. BLOCKER [ExecuteForkedTaskInsideNewJvm.java#L71](https://github.com/ow2-proactive/scheduling/blob/705e1a9a219af3a81a9a6c4971d867a27b680580/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java#L71): Remove this throw statement from this finally block. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1143)
1. MAJOR [ExecuteForkedTaskInsideNewJvm.java#L71](https://github.com/ow2-proactive/scheduling/blob/705e1a9a219af3a81a9a6c4971d867a27b680580/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java#L71): Refactor this code to not throw exceptions in finally blocks. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1163)
1. MINOR [KeyPairUtil.java#L39](https://github.com/ow2-proactive/scheduling/blob/705e1a9a219af3a81a9a6c4971d867a27b680580/common/common-client/src/main/java/org/ow2/proactive/authentication/crypto/KeyPairUtil.java#L39): Remove this unused import 'com.google.common.primitives.Longs'. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [ForkedTaskExecutor.java#L44](https://github.com/ow2-proactive/scheduling/blob/705e1a9a219af3a81a9a6c4971d867a27b680580/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/ForkedTaskExecutor.java#L44): Remove this unused import 'org.objectweb.proactive.extensions.processbuilder.exception.NotImplementedException'. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [ExecuteForkedTaskInsideNewJvm.java#L53](https://github.com/ow2-proactive/scheduling/blob/705e1a9a219af3a81a9a6c4971d867a27b680580/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java#L53): Define a constant instead of duplicating this literal "Context file " 3 times. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1192)
1. INFO [ForkedJvmTaskExecutionCommandCreator.java#L95](https://github.com/ow2-proactive/scheduling/blob/705e1a9a219af3a81a9a6c4971d867a27b680580/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ForkedJvmTaskExecutionCommandCreator.java#L95): Complete the task associated to this TODO comment. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1135)

**Comment by fviale** [comment]
Retest this please

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 4 issues

1. MINOR [KeyPairUtil.java#L39](https://github.com/ow2-proactive/scheduling/blob/4141c8c4e2426915569f5f39190276bd3c48ab30/common/common-client/src/main/java/org/ow2/proactive/authentication/crypto/KeyPairUtil.java#L39): Remove this unused import 'com.google.common.primitives.Longs'. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [ForkedTaskExecutor.java#L44](https://github.com/ow2-proactive/scheduling/blob/4141c8c4e2426915569f5f39190276bd3c48ab30/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/ForkedTaskExecutor.java#L44): Remove this unused import 'org.objectweb.proactive.extensions.processbuilder.exception.NotImplementedException'. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [ExecuteForkedTaskInsideNewJvm.java#L53](https://github.com/ow2-proactive/scheduling/blob/4141c8c4e2426915569f5f39190276bd3c48ab30/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java#L53): Define a constant instead of duplicating this literal "Context file " 3 times. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1192)
1. INFO [ForkedJvmTaskExecutionCommandCreator.java#L95](https://github.com/ow2-proactive/scheduling/blob/4141c8c4e2426915569f5f39190276bd3c48ab30/scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ForkedJvmTaskExecutionCommandCreator.java#L95): Complete the task associated to this TODO comment. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1135)

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/context/TaskContextSerializer.java [hunk 1]
```diff
@@ -59,6 +60,9 @@ public File serializeContext(TaskContext context, File directory) throws IOExcep
         try (ObjectOutputStream objectOutputStream = new ObjectOutputStream(new FileOutputStream(file))) {
             objectOutputStream.writeObject(context);
         }
+        if (context.isRunAsUser()) {
+            ForkerUtils.setSharedPermissions(file, false);
```
Comment (marcocast) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/context/TaskContextSerializer.java]: could you create a separate method more expressive ? something like setSharedExecutablePermissions

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java [hunk 2]
```diff
@@ -43,17 +43,42 @@
...
+            throw new IllegalStateException(CONTEXT_FILE + f + " exists but cannot be written");
+        }
         try (ObjectInputStream inputStream = new ObjectInputStream(new FileInputStream(f))) {
             return (TaskContext) inputStream.readObject();
         } finally {
+            deleteContextFile(f);
+        }
+    }
+
+    private static void deleteContextFile(File f) throws InterruptedException {
```
Comment (marcocast) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java]: can you change the f to something more readable

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java [hunk 3]
```diff
@@ -43,17 +43,42 @@
...
+        int cpt = 0;
```
Comment (marcocast) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java]: better name for cpt?

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java [hunk 4]
```diff
@@ -43,17 +43,42 @@
 
 public class ExecuteForkedTaskInsideNewJvm {
 
+    public static final int MAX_CONTEXT_WAIT = 6000;
+
+    public static final String CONTEXT_FILE = "Context file ";
+
     // 2 called by forked
     private static TaskContext deserializeContext(String pathToFile) throws IOException,
```
Comment (marcocast) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java]: check if is possible to remove the static private methods and have a simple test for this class

**Comment by fviale** [comment]
Retest this please

**Pull request merged**